base/daemon/BaseDaemon.cpp

```diff
@@ -152,7 +152,7 @@ static void signalHandler(int sig, siginfo_t * info, void * context)
     if (sig != SIGTSTP) /// This signal is used for debugging.
     {
         /// The time that is usually enough for separate thread to print info into log.
-        sleepForSeconds(10);
+        sleepForSeconds(20);  /// FIXME: use some feedback from threads that process stacktrace
```

Comment:
Ok.

About the feedback.
There are cases when multiple threads receive signals in short period of time (example: segfaults in all query processing threads). We cannot terminate after printing only the first stack trace. Some special logic with timeouts required.